backend/tests/expenses/test_expense_service.py

```
@@ -0,0 +1,144 @@
+    assert True  # Placeholder assertion
+
+@pytest.mark.asyncio
+async def test_settlement_algorithm_advanced():
+    """Test advanced settlement algorithm with graph optimization"""
+    
+    # Test scenario:
+    # A owes B $100
+    # B owes C $100
+    # Expected optimized: A pays C $100 directly
+    
+    user_balances = {
```

Ruff (0.11.9)

Local variable `user_balances` is assigned to but never used

Remove assignment to unused variable `user_balances`

(F841)